kie-server-parent/kie-server-controller/kie-server-controller-impl/src/main/java/org/kie/server/controller/impl/storage/InMemoryKieServerTemplateStorage.java

```diff
@@ -54,17 +60,44 @@ public ServerTemplate store(ServerTemplate serverTemplate) {
 
     @Override
     public List<ServerTemplate> load() {
-        return new ArrayList<ServerTemplate>(store.values());
+        synchronized (store) {
+            return store.values().stream().map(this::cloneServerTemplate).collect(Collectors.toList());
+        }
     }
 
     @Override
     public ServerTemplate load(String identifier) {
-        return store.get(identifier);
+        synchronized (store) {
+            return cloneServerTemplate(store.get(identifier));
+        }
     }
 
     @Override
     public boolean exists(String identifier) {
-        return store.containsKey(identifier);
+        synchronized (store) {
+            return store.containsKey(identifier);
+        }
+    }
+
+    private ServerTemplate cloneServerTemplate(ServerTemplate current) {
+
+        if (current == null) {
+            return null;
+        }
+
+        List<ContainerSpec> specs = current.getContainersSpec().stream().map(ContainerSpec::new).collect(Collectors.toList());
+
+        ServerTemplate serverTemplate = new ServerTemplate(current.getId(),
+                                                           current.getName(),
+                                                           new ArrayList<>(current.getCapabilities()),
+                                                           current.getConfigs().isEmpty() ? new EnumMap<>(Capability.class) : new EnumMap<>(current.getConfigs()),
```

Comment:
Here we store the `EnumMap` instance but in the original state of an object there is an instance of `HashMap` created. https://github.com/kiegroup/droolsjbpm-integration/blob/1b42ad1f1009442f13a500842c97b16e19d06648/kie-server-parent/kie-server-controller/kie-server-controller-api/src/main/java/org/kie/server/controller/api/model/spec/ServerTemplate.java#L40